src/main/java/io/github/oliviercailloux/teach_spreadsheets/CoursePref.java

```diff
@@ -0,0 +1,208 @@
+package io.github.oliviercailloux.teach_spreadsheets;
+
+import com.google.common.base.MoreObjects;
+
+/**
+ * Class used to store a teacher's preference for one course.
+ * Uses Builder pattern implementation.
+ * @see https://codereview.stackexchange.com/questions/127391/simple-builder-pattern-implementation-for-building-immutable-objects/127509#127509
+ */
+public class CoursePref {
+	private static final String EXCEPTION = "A preferred number of groups needs to be positive.";
+	
+	private Course course;
+	private Teacher teacher;
+	
+	/**
+	 * Degree of preference of teacher for course, represented by chars
```

Comment:
Inélégant : seuls A, B, C ou ' ' sont permis et on ne s’en rend pas compte facilement, et on ne comprend pas nécessairement ce que ' ' veut dire. Créer un enum avec ces trois possibilités plus UNSPECIFIED et utiliser ce type pour les prefs.